src/Microsoft.Sbom.Api/Executors/Spdx30SerializationStrategy.cs

```diff
@@ -15,81 +14,97 @@ namespace Microsoft.Sbom.Api.Workflows.Helpers;
     {
         elementsSupportingConfigs.Add(config);
-        return true;
+        return false;
     }
 
-    public async Task<List<FileValidationResult>> WriteJsonObjectsToSbomAsync(
-        ISbomConfig sbomConfig,
-        string spdxManifestVersion,
-        IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator,
-        IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator,
-        IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator,
-        IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator)
+    public void AddMetadataToSbom(ISbomConfigProvider sbomConfigs, ISbomConfig config)
     {
-        fileArrayGenerator.SbomConfig = sbomConfig;
-        packageArrayGenerator.SbomConfig = sbomConfig;
-        relationshipsArrayGenerator.SbomConfig = sbomConfig;
-        externalDocumentReferenceGenerator.SbomConfig = sbomConfig;
-
-        fileArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
-        packageArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
-        relationshipsArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
-        externalDocumentReferenceGenerator.SpdxManifestVersion = spdxManifestVersion;
-
-        // Holds the SPDX IDs of all the elements that have been written to the SBOM. Used for deduplication.
-        var elementsSpdxIdList = new HashSet<string>();
-
-        WriteContext(sbomConfig);
-
-        sbomConfig.JsonSerializer.StartJsonArray(Constants.SPDXGraphHeaderName);
-
-        // Files section
-        var generationResult = await fileArrayGenerator.GenerateAsync();
-        WriteElementsToSbom(generationResult, elementsSpdxIdList);
-
-        // Packages section
-        var packagesGenerationResult = await packageArrayGenerator.GenerateAsync();
-        generationResult.Errors.AddRange(packagesGenerationResult.Errors);
-        WriteElementsToSbom(packagesGenerationResult, elementsSpdxIdList);
-
-        // External Document Reference section
-        var externalDocumentReferenceGenerationResult = await externalDocumentReferenceGenerator.GenerateAsync();
-        generationResult.Errors.AddRange(externalDocumentReferenceGenerationResult.Errors);
-        WriteElementsToSbom(externalDocumentReferenceGenerationResult, elementsSpdxIdList);
-
-        // Relationships section
-        var relationshipGenerationResult = await relationshipsArrayGenerator.GenerateAsync();
-        generationResult.Errors.AddRange(relationshipGenerationResult.Errors);
-        WriteElementsToSbom(relationshipGenerationResult, elementsSpdxIdList);
+        // Not supported for SPDX 3.0 and above, only supported for SPDX 2.2.
+        // For SPDX 3.0 and above, the metadata is added as part of the SpdxDocument and CreationInfo elements.
+    }
 
-        sbomConfig.JsonSerializer.EndJsonArray();
+    public void StartGraphArray(IList<ManifestInfo> manifestInfosFromConfig, ISbomConfigProvider sbomConfigs)
```

Comment:
Question: If the `ISerializationStrategy` is already specific to an ISbomConfig, why do we need to iterate through all of the manifests?

Reply:
This implementation has since been changed, and should address the question as well